pkg/kubectl/resource/builder.go

```diff
@@ -587,7 +587,7 @@ func (b *Builder) visitorResult() *Result {
 		return &Result{singular: singular, visitor: visitors, sources: b.paths}
 	}
 
-	return &Result{err: fmt.Errorf("you must provide one or more resources by argument or filename")}
+	return &Result{err: fmt.Errorf(`you must provide one or more resources by argument or filename, try "kubectl help [command]" for help`)}
```

Comment:
This check probably belongs in cmdutil.CheckErr (which is also not Kubectl specific).  We don't do enough delegation today to allow the check to be handled by the factory, but that's ultimately the right place to do so.